libraries/rush-lib/src/logic/operations/PeriodicCallback.ts

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+export type ICallbackFn = () => Promise<void> | void;
+
+export interface IPeriodicCallbackOptions {
+  interval: number;
+}
+
+/**
+ * A help class to run callbacks in a loop with a specified interval.
+ *
+ * @beta
+ */
+export class PeriodicCallback {
+  private _callbacks: ICallbackFn[];
+  private _interval: number;
+  private _timeoutId: NodeJS.Timeout | undefined;
+  private _isRunning: boolean;
+
+  public constructor(options: IPeriodicCallbackOptions) {
+    this._callbacks = [];
+    this._interval = options.interval;
+    this._isRunning = false;
+  }
+
+  public addCallback(callback: ICallbackFn): void {
+    if (this._isRunning) {
+      throw new Error('Can not add callback while watcher is running');
+    }
+    this._callbacks.push(callback);
+  }
+
+  public start(): void {
+    if (this._timeoutId) {
+      throw new Error('Watcher already started');
+    }
+    if (this._callbacks.length === 0) {
+      return;
+    }
+    this._isRunning = true;
+    this._timeoutId = setTimeout(() => {
```

Comment:
Can just use `setInterval`/`clearInterval` instead of `setTimeout`/`clearTimeout` to get periodic invocation.

Reply:
I take it a second thought. Originally, I always choose "setTimeout" over "setInterval" when writing JS code.  "setTimeout" is more flexible to control the next call. while "setInterval" seems more naively repeat doing things. I will try to use "setInterval" here, maybe "setInterval" is slightly better in this simple case. :)